asmcomp/cmmgen.ml

```diff
@@ -1346,26 +1355,23 @@ let make_switch arg cases actions dbg =
   let is_const = function
     | Cconst_int n
     | Cconst_pointer n -> (n land 1) = 1
-    | Cconst_natint _
-    | Cconst_float _
+    | Cconst_natint n
+    | Cconst_natpointer n -> (Nativeint.(to_int (logand n one) = 1))
```

Comment:
> I left in the low-bit check: I'm not convinced that it's correct to allow Cload to return an int with a zero in the low bit.

Note that this is probably what happens in [strmatch.ml](https://github.com/ocaml/ocaml/blob/trunk/asmcomp/strmatch.ml#L76-L78), which comes from [MPR#6269](https://caml.inria.fr/mantis/view.php?id=6269). But I am completely fine with no trying it here -- I'm not sure why it would or wouldn't be correct either.